api/v1alpha1/ndb_api_helpers.go

```diff
@@ -155,43 +156,120 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
+	log.Info("Entered ndb_api_helpers.GetProfiles", "Input profiles", inputProfiles)
+
+	allProfiles, err := GetAllProfiles(ctx, ndbclient)
 	if err != nil {
+		log.Error(err, "Profiles could not be fetched")
 		return
 	}
 
-	genericProfiles := util.Filter(profiles, func(p ProfileResponse) bool { return p.EngineType == DATABASE_ENGINE_TYPE_GENERIC })
-	dbEngineSpecificProfiles := util.Filter(profiles, func(p ProfileResponse) bool { return p.EngineType == GetDatabaseEngineName(dbType) })
-
-	// Specifying the usage of small compute profiles
-	computeProfiles := util.Filter(genericProfiles, func(p ProfileResponse) bool {
-		return p.Type == PROFILE_TYPE_COMPUTE && strings.Contains(strings.ToLower(p.Name), "small")
-	})
-	storageProfiles := util.Filter(genericProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_STORAGE })
-	// Specifying the usage of single instance topoplogy
-	softwareProfiles := util.Filter(dbEngineSpecificProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_SOFTWARE && p.Topology == TOPOLOGY_SINGLE })
-	networkProfiles := util.Filter(dbEngineSpecificProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_NETWORK })
-	dbParamProfiles := util.Filter(dbEngineSpecificProfiles, func(p ProfileResponse) bool { return p.Type == PROFILE_TYPE_DATABASE_PARAMETER })
-
-	// Some profile not found, return an error
-	if len(computeProfiles) == 0 || len(softwareProfiles) == 0 || len(storageProfiles) == 0 || len(networkProfiles) == 0 || len(dbParamProfiles) == 0 {
-		err = errors.New("oob profile: one or more OOB profile(s) were not found")
-		return
+	dbEngineSpecific := util.Filter(allProfiles, func(p ProfileResponse) bool { return p.EngineType == GetDatabaseEngineName(instanceSpec.Type) })
+
+	// Compute Profile
+	compute, err := inputProfiles.Compute.Resolve(ctx, allProfiles, ComputeOOBProfileResolver)
+	if err != nil {
+		log.Error(err, "Compute Profile could not be resolved", "Input Profile", inputProfiles.Compute)
+		return nil, err
+	}
+
+	// Software Profile
+	software, err := inputProfiles.Software.Resolve(ctx, dbEngineSpecific, SoftwareOOBProfileResolver)
+	if err != nil {
+		log.Error(err, "Software Profile could not be resolved", "Input Profile", inputProfiles.Software)
+		return nil, err
+	}
+
+	// Network Profile
+	networkProfile, err := inputProfiles.Network.Resolve(ctx, dbEngineSpecific, NetworkOOBProfileResolver)
+	if err != nil {
+		log.Error(err, "Network Profile could not be resolved", "Input Profile", inputProfiles.Network)
+		return nil, err
 	}
 
-	profileMap[PROFILE_TYPE_COMPUTE] = computeProfiles[0]
-	profileMap[PROFILE_TYPE_STORAGE] = storageProfiles[0]
-	profileMap[PROFILE_TYPE_SOFTWARE] = softwareProfiles[0]
-	profileMap[PROFILE_TYPE_NETWORK] = networkProfiles[0]
-	profileMap[PROFILE_TYPE_DATABASE_PARAMETER] = dbParamProfiles[0]
+	// DB Param Profile
+	dbParamProfile, err := inputProfiles.DbParam.Resolve(ctx, dbEngineSpecific, DbParamOOBProfileResolver)
```

Comment:
Can you also remove the profile suffix from these like you did for compute and software? :D

Reply:
Done